core/src/CMakeLists.txt

```diff
@@ -46,20 +46,23 @@ endif ()
 
 link_directories(${CMAKE_BINARY_DIR}/lib)
 
-set(LIB_BUILD_TYPE SHARED)
-
 file(GLOB_RECURSE SRC_FILES *.cpp)
 file(GLOB_RECURSE HEADERS_FILES *.h *.hpp)
 list(REMOVE_ITEM SRC_FILES ${ledger-core-jni-sources})
+list(REMOVE_ITEM SRC_FILES dummy.cpp)
 list(REMOVE_ITEM HEADERS_FILES ${ledger-core-jni-sources})
 
-add_library(ledger-core ${LIB_BUILD_TYPE}
-            ${ledger-core-sources}
-            ${SRC_FILES}
-            ${HEADERS_FILES})
+add_library(ledger-core-interface INTERFACE)
+
+target_compile_definitions(ledger-core-interface INTERFACE ledger_core_EXPORTS)
 
 string(FIND "${CMAKE_OSX_SYSROOT}" "iphone" IS_IOS)
 if(IS_IOS GREATER_EQUAL 0)
+	add_library(ledger-core SHARED
```

Comment:
So for iOS, `ledger-core` is a shared library but for the rest will be object libraries. Why?

Reply:
No, ledger-core will always be a shared library. For other platforms I am creating temporary "object library" ledger-core-obj from which ledger-core and ledger-core-static are build.